CougSat1-ADCS/src/interfaces/CDH.cpp

```diff
@@ -1,31 +1,66 @@
 #include "CDH.h"
 
-#include <CISConsole.h>
-
 /**
  * @brief Construct a new CDH::CDH object
  *
  * @param i2c connected to the C&DH
  */
-CDH::CDH(I2CSlave & i2c) : i2c(i2c) {}
+CDH::CDH(uint8_t addr, PinName sda, PinName scl) : i2c(sda, scl) 
+{
+  i2c.address(addr);
+}
 
 /**
  * @brief Checks if a message is available to read
  *
  * @return true is the PMIC is addressed
  */
-bool CDH::hasMessage() {
+bool CDH::messageReceived() {
   int state = i2c.receive();
-  return state == I2CSlave::ReadAddressed || state == I2CSlave::WriteAddressed;
+  return state == mbed::I2CSlave::WriteAddressed ;
 }
 
 /**
- * @brief Process the message coming from the C&DH
+ * @brief Checks if a message is requested by master
  *
- * @return mbed_error_code_t error code
+ * @return true is the PMIC is addressed
  */
-mbed_error_code_t CDH::processMessage() {
-  // Get message
-  // Switch on commandID and fire off the proper function
-  return MBED_ERROR_UNSUPPORTED;
+bool CDH::messageRequested() {
+  int state = i2c.receive();
+  return state == mbed::I2CSlave::ReadAddressed ;
 }
+
+/**
+ * @brief Reads message from I2CSlave buffer into var message
+ * 
+ */
+void CDH::readCDH()
+{
+  printf("Reading i2c buffer\n Messsage = ");
+  i2c.read(message, MESSAGELENGTH);
+  for (uint8_t i = 0; message[i] != NULL; i++)
+  {
+      printf("%c,", message[i]);
+  }
+  printf("\n");
+}
+
+/**
+ * @brief Getter for private message array
+ * 
+ */
+char* CDH::getMessage()
+{
+  return message;
```

Comment:
What is the function of this class if it just wraps i2cSlave? I thought the class should process the messages and set flags or enqueue tasks based on that info

Reply:
Our current design has the message data being held in the command and control wrapper, so it can easily pass it to other subsystems. The CDH class is in charge of sending messages and receiving them but does not know how to process them as it would be very inefficient.